cmd/kube-scheduler/app/server.go

```diff
@@ -292,7 +297,6 @@ func Run(ctx context.Context, cc *schedulerserverconfig.CompletedConfig, sched *
 				case <-ctx.Done():
 					// We were asked to terminate. Exit 0.
 					logger.Info("Requested to terminate, exiting")
-					os.Exit(0)
 				default:
 					// We lost the lock.
 					logger.Error(nil, "Leaderelection lost")
```

Comment:
Any way to distinguish from here?

Reply:
@Jefftree 
If you know the way to distinguish from here, I want to know about it too because I'm not familiar with the implementation of coordinated leader election.